### PR TITLE
Show restart count warnings only for latest deployment

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/10417

@kargakis PTAL

### CODE REVIEW

**Comment by mfojtik** [comment]
[test]

File: pkg/cmd/cli/describe/projectstatus.go [hunk 1]
```diff
@@ -937,6 +937,8 @@ func describeDeployments(f formatter, dcNode *deploygraph.DeploymentConfigNode,
 	out := []string{}
 	deploymentsToPrint := append([]*kubegraph.ReplicationControllerNode{}, inactiveDeployments...)
 
+	latestName := deployutil.LatestDeploymentNameForConfig(dcNode.DeploymentConfig)
```
Comment (kargakis) [inline pkg/cmd/cli/describe/projectstatus.go]: Move this just before the loop you are using it

File: pkg/cmd/cli/describe/projectstatus.go [hunk 2]
```diff
@@ -950,7 +952,11 @@ func describeDeployments(f formatter, dcNode *deploygraph.DeploymentConfigNode,
 	}
 
 	for i, deployment := range deploymentsToPrint {
-		out = append(out, describeDeploymentStatus(deployment.ReplicationController, i == 0, dcNode.DeploymentConfig.Spec.Test, restartCount))
+		if latestName == deployment.ReplicationController.Name {
+			out = append(out, describeDeploymentStatus(deployment.ReplicationController, i == 0, dcNode.DeploymentConfig.Spec.Test, restartCount))
+		} else {
+			out = append(out, describeDeploymentStatus(deployment.ReplicationController, i == 0, dcNode.DeploymentConfig.Spec.Test, 0))
```
Comment (kargakis) [inline pkg/cmd/cli/describe/projectstatus.go]: If this deployment is active ie. has pods but is about to be rolled out, I would want to know if its pods have restarted. I think we need the restart count to be part of the RC node.

Comment (mfojtik) [inline pkg/cmd/cli/describe/projectstatus.go]: @kargakis I can add "or is active deployment"

Comment (kargakis) [inline pkg/cmd/cli/describe/projectstatus.go]: Don't you need to have a different restart count though?

**Comment by mfojtik** [comment]
@kargakis fixed PTAL

File: pkg/cmd/cli/describe/projectstatus.go (hunk 2 above)
Comment (mfojtik) [inline pkg/cmd/cli/describe/projectstatus.go]: @kargakis maybe I don't understand your original question... let say you have active deployment that have pods (active means it is currently being rolled out, right?) and you want to know if it has pods are restarting... so when the deployment complete, you will see the restarts, correct? If it fails, then it means you will see failed deployment (or we revert you to previous deployment).

**Comment by mfojtik** [comment]
@kargakis PTAL

I used different approach, passing the func will allow to get the count for every deployment. I tested this with paused deployment and it works (I can see the pods restarting in "old" deployment).

**Comment by mfojtik** [comment]
[test]

https://github.com/openshift/origin/issues/9624

**Comment by kargakis** [comment]
> I used different approach, passing the func will allow to get the count for every deployment. I tested this with paused deployment and it works (I can see the pods restarting in "old" deployment).

Can't you just execute the function and pass the count?

**Comment by mfojtik** [comment]
@kargakis no because I need the graph

**Comment by kargakis** [comment]
LGTM [merge]

**Pull request merged**